daemon/main.c

```diff
@@ -627,6 +623,45 @@ main_save_scheduler(void)
     sparam_save = sparam;
 }
 
+//
+//  main_set_scheduler
+//
+//  Set scheduling policy and priority of the process.
+//  Then all the threads will use the same policy and priority.
+//
+
+MTC_STATIC void
+main_set_scheduler(void)
+{
+    struct sched_param sparam = {0};
+    int sched_policy;
+    int status;
+
+    move_to_root_cgroup();
+
+    // set scheduler policy and priority
+    sched_policy = SCHED_RR;
+    // Some of kernel tasks are running at priority 50 (e.g. the primary IRQ
+    // handler threads) and priority 49 (e.g. the secondary IRQ handler
+    // threads). Set the priority of xHA as 48 as high as possible but less
+    // than these kernel tasks to avoid priority inversion between xHA and
+    // IRQ handlers.
+    sparam.sched_priority = 48;
+
+    status = sched_setscheduler(getpid() , sched_policy, &sparam);
+    if (status == 0) 
+    {
+        log_message(MTC_LOG_DEBUG, "HA daemon set scheduler policy=%d priority=%d\n", 
+                    sched_policy, sparam.sched_priority);
+    }
+    else 
+    {
+        log_message(MTC_LOG_WARNING, "HA daemon set scheduler failed (sys %d) status=%d policy=%d priority=%d\n", 
```

Comment:
Would this need to exit as well?  As:
1. the `sched_setscheduler` is not expected to fail, and
2. and this error is expected to be caught when the platform is changed in the future.

Reply:
Added `main_exit`.